pandas/tests/indexes/common.py

```diff
@@ -937,3 +937,14 @@ def test_empty(self):
         index = self.create_index()
         assert not index.empty
         assert index[:0].empty
+
+    @pytest.mark.parametrize('how', ['outer', 'inner', 'left', 'right'])
+    def test_join_self(self, how):
+        index = self.create_index()
+        joined = index.join(index, how=how)
+
+        if index is index.unique():
```

Comment:
Using `index.is_unique` will cause this test to fail in the  event that an index `ix` is unique but `ix is ix` returns `False`.  That may be a bug (it's happening with `TimedeltaIndex` currently)?